src/zmalloc.c

```diff
@@ -184,15 +184,21 @@ void *ztrycalloc_usable(size_t size, size_t *usable) {
 /* Allocate memory and zero it or panic.
  * We need this wrapper to have a calloc compatible signature */
 void *zcalloc_num(size_t num, size_t size) {
+    /* Ensure that the arguments to calloc(), when multiplied, do not wrap.
+     * Division operations are susceptible to divide-by-zero errors so we also check it. */
+    if ((size == 0) || (num > UINTMAX_MAX/size)) {
+        zmalloc_oom_handler(num, size);
```

Comment:
Not sure we need to call `zmalloc_oom_handler`, according to the `calloc` docs passing an overflowing or 0 value should simply return `NULL`.
Also I don't see value in adding the extra `num` and `size` to the logging.

Reply:
> Not sure we need to call zmalloc_oom_handler, according to the calloc docs passing an overflowing or 0 value should simply return NULL.

@yoav-steinberg I believe we must call oom_handler within the redis context even tough it's not required by the calloc definition. @antirez addressed issue https://github.com/redis/redis/issues/509 specifically with the oom_handler so that we can log with further details and exit when we cannot allocate the required memory:
Quoting him:
> The result was that in the case of an actual out of memory in Redis
where malloc returned NULL (In Linux this actually happens under
specific overcommit policy settings and/or with no or little swap
configured) the error was not properly logged in the Redis log.

With that in mind, IMO, we should call the handler and return NULL. In that manner, we have a configurable way of deciding wether to abort on log or just expect the caller of zcalloc to handle it. agree?


With regards to:
>  Also I don't see value in adding the extra num and size to the logging.

I'm reverting and logging `SIZE_MAX` on the scenario of overflow as suggested by @oranagra .

Reply:
you're right. thanks.